src/main/java/org/venice/beachfront/controllers/ShapefileFromJSONController.java

```diff
@@ -0,0 +1,82 @@
+package org.venice.beachfront.controllers;
+
+import java.io.PrintWriter;
+import java.io.StringWriter;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.ResponseBody;
+import org.venice.beachfront.services.ShapefileConverter;
+
+@Controller
+public class ShapefileFromJSONController {
+	/**
+	 * Injected dependency on the Shapefile converter service.
+	 */
+	private ShapefileConverter shapefileConverter;
+	
+	/**
+	 * Construct the ShapefileFromJSONFileController.
+	 * 
+	 * @param piazzaApi
+	 *            The Piazza API service instance to use
+	 * @param shapefileConverter
+	 *            The ShapefileConverter service instance to use
+	 */
+	@Autowired
+	public ShapefileFromJSONController(ShapefileConverter shapefileConverter) {
+		this.shapefileConverter = shapefileConverter;
+	}
+
+	/**
+	 * Core handler for converting from a Piazza ID to a GeoJSON file.
+	 * 
+	 * This controller provides a simple interface for immediate conversion of 
+	 * GeoJSON data to a Shapefile (.zip containing .shp, etc.).
+	 * 
+	 * {@link RequestMapping} decorated using the path {@code /convertshp},
+	 * request method POST, and as producing "application/zip". 
+	 * 
+	 * @param geojson a byte array containing the raw GeoJSON received via POST
+	 * request body
+	 * 
+	 * @return A {@link ResponseEntity} that contains the converted Shapefile data
+	 */
+	@RequestMapping(path = "/convertshp", method = RequestMethod.POST, consumes = {"application/json"}, produces = {"application/zip"} )
+	@ResponseBody
+	public ResponseEntity<byte[]> convertToShapefile(@RequestBody byte[] geojson) {
+		byte[] gpkg = this.shapefileConverter.apply(geojson);
+		HttpHeaders headers = new HttpHeaders();
+		headers.setContentType(new MediaType("application", "application/zip"));
```

Comment:
Should be `new MediaType("application", "zip")`. The subtype of the media type is not allowed to contain a slash.